### PR TITLE
Update hysteria

### PR DESCRIPTION
**Pull request by stevenmcdonald**
This updates Hysteria to v1.3.5. Go >=1.19 is required to build it

I guess we'll call this 1.4.3?

AAR is here: https://github.com/stevenmcdonald/IEnvoyProxy/releases/tag/e1.4.3-alpha-1

### CODE REVIEW

**Comment by stevenmcdonald** [comment]
FWIW I tested it out with my go test project, and all the services seemed to work with the production servers

**Review by mnbogner — APPROVED** [review]
nothing jumps out at me in the diff and it works when i build envoy/wiki unblocked so i approve

File: IEnvoyProxy/go.mod [hunk 1]
```diff
@@ -1,19 +1,164 @@
...
 require (
 	git.torproject.org/pluggable-transports/snowflake.git/v2 v2.0.0-00010101000000-000000000000
-	github.com/tobyxdd/hysteria v1.0.5
+	github.com/apernet/hysteria/app v1.3.5
 	github.com/v2fly/v2ray-core v4.45.2+incompatible
-	github.com/v2fly/v2ray-core/v5 v5.0.7 // indirect
-	golang.org/x/mobile v0.0.0-20230427221453-e8d11dd0ba41 // indirect
+)
+
+require (
```
Comment (vitriolix) [inline IEnvoyProxy/go.mod:11]: that's quite a require list there o_O

**Comment by vitriolix** [comment]
aar builds for me an works in GFD, 🚢

File: IEnvoyProxy/go.mod (hunk 1 above)
Comment (stevenmcdonald) [reply IEnvoyProxy/go.mod:11]: Yeah, Hysteria added a bunch of dependencies: https://github.com/apernet/hysteria/blob/master/app/go.mod

**Pull request merged**